core/src/main/java/com/graphhopper/PathWrapper.java

```diff
@@ -243,6 +244,26 @@ public void setInstructions(InstructionList instructions) {
         this.instructions = instructions;
     }
 
+    public void addPathDetails(List<PathDetails> details) {
+        if (!this.pathDetails.isEmpty() && !details.isEmpty() && this.pathDetails.size() != details.size()) {
+            throw new IllegalStateException("Details have to be the same size");
+        }
+        for (int i = 0; i < details.size(); i++) {
+            PathDetails pd;
+            if (this.pathDetails.containsKey(details.get(i).getName())) {
+                pd = this.pathDetails.get(details.get(i).getName());
+                pd.merge(details.get(i));
+            } else {
```

Comment:
Ups no, forget about it :) but I think there will be either all merge or all put operations?

BTW: here is also no need to call pathDetails.put for the merge case